pkg/proxy/endpointslicecache.go

```diff
@@ -122,16 +126,21 @@ func newEndpointSliceInfo(endpointSlice *discovery.EndpointSlice, remove bool) *
 
 	if !remove {
 		for _, endpoint := range endpointSlice.Endpoints {
-			if endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready {
-				eInfo := endpointInfo{
-					Addresses: endpoint.Addresses,
-					Topology:  endpoint.Topology,
-				}
-				if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceNodeName) {
-					eInfo.NodeName = endpoint.NodeName
-				}
-				esInfo.Endpoints = append(esInfo.Endpoints, &eInfo)
+			epInfo := &endpointInfo{
```

Comment:
While this may look like a change in scale (how many endpoints are included), the number of unready/terminating/not- serving endpoints should be O(1) at any time and a small subset of ready endpoints, except when things like zone failures occur or a very aggressive rollout of the deployment fails, which still is O(N) endpoints and likel at most 2x the number of previous endpoints (if you have a deployment with 10k pods, the most pods you can have in this state is roughly 20k, so while you're taking a linear hit in memory, if you would fail at 20k pods you would likely fail at 15k too).

So I think this is good from a scale perspective, but operators may see a tick upwards in 1.21/1.22 because of this, but that was also expected for edpnointslices.